src/main.rs

```diff
@@ -1,126 +1,267 @@
+) {
+    let scale = 16;
+    let background = rgba_to_bgra_u32(Rgba {r: 0.62, g: 0.62, b: 0.62, a: 1.0});
 
-    pub fn edge(&self, eid: EdgeId) -> (Point2D, Point2D) {
-        unimplemented!();
-    }
+    let mut color = Buffer2d::new([image_width, image_height], background);
+    let mut depth = Buffer2d::new([image_width, image_height], 1.0);
 
-    pub fn face(&self, fid: usize) -> Vec<Point2D> {
-        unimplemented!();
-    }
+    // Scaled screen buffer
+    let (screen_width, screen_height) = (image_width * scale, image_height * scale);
+    let mut screen = Buffer2d::new([screen_width, screen_height], 0);
 
-    //TODO: Lookup "impl trait" syntax and learn about iterators
-    pub fn vertices(&self) -> impl Iterator<Item=(VertexId, Point2D)> {
-        unimplemented!();
-    }
+    let (axis_width, axis_height) = (128, 128);
+    let axis = if screen_width > axis_width && screen_height > axis_height {
+        let mut axis_color = Buffer2d::new([axis_width, axis_height], background);
+        render_axis(&mut axis_color, view);
+        Some(axis_color)
+    } else { None };
 
-    pub fn edges(&self) -> impl Iterator<Item=(EdgeId, (Point2D, Point2D))> {
-        unimplemented!();
-    }
+    let mut win = minifb::Window::new(
+        "Test Project",
+        image_width * scale,
+        image_height * scale,
+        minifb::WindowOptions::default()
+    ).unwrap();
 
-    pub fn faces(&self) -> impl Iterator<Item=(FaceId, Vec<Point2D>)> {
-        unimplemented!();
+    // Keep the program from ending
+    let mut i = 0;
+    while win.is_open() && !win.is_key_pressed(Key::Escape, KeyRepeat::No) {
+        color.clear(background);
+        depth.clear(1.0);
+
+        let meshes = &frames[i];
+        render(&mut color, &mut depth, model, view, projection, meshes, 0.15);
+
+        scale_buffer(&mut screen, &color);
+
+        if let Some(axis) = &axis {
+            // Unsafe because we are guaranteeing that the provided offset is not out of bounds
+            unsafe { copy(&mut screen, &axis, (0, screen_height - axis_height)); }
+        }
+
+        win.update_with_buffer(screen.as_ref()).unwrap();
+
+        // No need to use 100% CPU for no reason
+        thread::sleep(Duration::from_millis(1000 / 10));
+
+        i = (i + 1) % frames.len();
     }
 }
 
-#[derive(Debug, Clone)]
-struct Object {
-    id: ObjectId, //TODO
-    mesh: MeshId,
-    //TODO: See if we need to have a list of material IDs here
+/// Renders a set of axis that match the orientation of the given view matrix
+fn render_axis(
+    axis_color: &mut Buffer2d<u32>,
+    view: Mat4<f32>,
+) {
+    // Only want to load this once
+    thread_local! {
+        /// This is an example for using doc comment attributes
+        static AXIS_MESHES: Vec<Mesh> = load_frame("samples/axis/axis.obj");
+    }
+
+    let axis_size = axis_color.size();
+    let projection = Mat4::perspective_rh_no(0.35*PI, (axis_size[0] as f32)/(axis_size[1] as f32), 0.01, 100.0)
```

Comment:
[MINOR] If we have a camera type, we could replace the need to completely create a new projection matrix from scratch, and just create a mutator to adjust the vertical FOV parameter :) (since we create a projection matrix as well in the main function)